lib/payment.js

```diff
@@ -204,8 +204,15 @@ class Payment {
             _signature.set(p, json.seals.wallet.signature);
         }
         if (json.sender && json.sender.data && typeof json.sender.data === 'string') {
-            const senderData = JSON.parse(Buffer.from(json.sender.data, 'base64').toString('ascii'));
-            _senderRef.set(p, senderData.ref);
+            const jsonSenderData = Buffer.from(json.sender.data, 'base64').toString('ascii');
+            try {
+                const senderData = JSON.parse(jsonSenderData);
+                _senderRef.set(p, senderData.ref);
+            }
+            catch (e) {
+                // Ignore parse errors and leave the _senderRef undefined.
+                // TODO: Is there an exception safe alternative to JSON.parse?
```

Comment:
From what I know, there's no version of `JSON.parse` or similar, that that doesn't potentially throw. Would be nice to have thought, something like JSON.tryParse() :-)

Of course, there could be a function I'm not aware of also.